nexus/db-macros/outputs/update_artifact.txt

```diff
@@ -0,0 +1,206 @@
+    /// point, we need the `OpContext` and `DataStore` that are being
+    /// used for this lookup.
+    fn lookup_root(&self) -> &LookupPath<'a> {
+        match &self {
+            UpdateArtifact::Error(root, ..) => root.lookup_root(),
+            UpdateArtifact::PrimaryKey(root, ..) => root.lookup_root(),
+        }
+    }
+    /// Fetch the database row for a resource by doing a lookup by id
+    ///
+    /// This function checks whether the caller has permissions to read
+    /// the requested data.  However, it's not intended to be used
+    /// outside this module.  See `fetch_for(authz::Action)`.
+    async fn fetch_by_id_for(
+        opctx: &OpContext,
+        datastore: &DataStore,
+        v0: &String,
+        v1: &i64,
+        v2: &KnownArtifactKind,
+        action: authz::Action,
+    ) -> LookupResult<(authz::UpdateArtifact, nexus_db_model::UpdateArtifact)> {
+        let (authz_update_artifact, db_row) = Self::lookup_by_id_no_authz(
+                opctx,
+                datastore,
+                v0,
+                v1,
+                v2,
+            )
+            .await?;
+        opctx.authorize(action, &authz_update_artifact).await?;
+        Ok((authz_update_artifact, db_row))
+    }
+    /// Lowest-level function for looking up a resource in the database
+    /// by id
+    ///
+    /// This function does not check whether the caller has permission
+    /// to read this information.  That's why it's not `pub`.  Outside
+    /// this module, you want `fetch()` or `lookup_for(authz::Action)`.
+    async fn lookup_by_id_no_authz(
+        opctx: &OpContext,
+        datastore: &DataStore,
+        v0: &String,
+        v1: &i64,
+        v2: &KnownArtifactKind,
+    ) -> LookupResult<(authz::UpdateArtifact, nexus_db_model::UpdateArtifact)> {
+        use db::schema::update_artifact::dsl;
+        let db_row = dsl::update_artifact
+            .filter(dsl::name.eq(v0.clone()))
+            .filter(dsl::version.eq(v1.clone()))
+            .filter(dsl::kind.eq(v2.clone()))
+            .select(nexus_db_model::UpdateArtifact::as_select())
+            .get_result_async(&*datastore.pool_connection_authorized(opctx).await?)
+            .await
+            .map_err(|e| {
+                public_error_from_diesel(
+                    e,
+                    ErrorHandler::NotFoundByLookup(
+                        ResourceType::UpdateArtifact,
+                        LookupType::ByCompositeId(
+                            format!(
+                                "name = {:?}, version = {:?}, kind = {:?}", v0, v1, v2,
+                            ),
+                        ),
+                    ),
+                )
+            })?;
+        let authz_update_artifact = Self::make_authz(
+            &&authz::FLEET,
+            &db_row,
+            LookupType::ByCompositeId(
+                format!("name = {:?}, version = {:?}, kind = {:?}", v0, v1, v2,),
```

Comment:
Well, the tests are certainly working!

```suggestion
                format!("name = {:?}, version = {:?}, kind = {:?}", v0, v1, v2),
```

Should fix CI, I think

Reply:
Ummmm I can't repro this locally, the test passes for me

Reply:
This looks like prettyplease doing its thing? But why would it behave differently on my machine?

Reply:
Fixed by enabling the verbatim feature.

It looks like prettyplease is being built two different ways, once with the verbatim feature and once without. I don't know why hakari isn't catching that -- will diagnose this later.